kafka-streams-partitioners/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/MultiLevelPartitionerConfig.java

```diff
@@ -0,0 +1,63 @@
+package org.hypertrace.core.kafkastreams.framework.partitioner;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.partitioner.config.service.v1.PartitionerGroup;
+import org.hypertrace.partitioner.config.service.v1.PartitionerProfile;
+
+@Slf4j
+class MultiLevelPartitionerConfig {
```

Comment:
Naming: So this doesn't really have anything to do with multi-level, that's just how we want to use it right? This purely describes the config of the single top level - the weighted group based partitioning.

Reply:
Refactored as below. Let me know if you have better naming suggestions.
MultiLevelPartitionerConfig -> WeightedGroupProfile
PartitionGroupConfig -> WeightedGroup